test/tests/functional/pbs_basil_support.py

```diff
@@ -362,5 +407,393 @@ def test_hbmemm_rsc(self):
                         "requested hbmem of %d mb" %
                         (hbm_assig, v_name, hbm_req))
                     self.assertTrue(False)
+
+        # Delete the Job.
+        self.server.delete(job_id, wait=True)
+
+    def test_job_request_insufficent_hbmemm_rsc(self):
+        """
+        Submit a job request that requests more than available HBMEM.
+        Check if the job is in the 'Q' state with valid comment.
+        Delete the job
+        """
+
+        knl_vnodes = self.get_knl_vnodes()
+
+        if len(knl_vnodes) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (knl_vnodes))
+
+        hbm_req = 18000
+        a = {'Resource_List.select': '1:hbmem=%dmb' % hbm_req}
+        job = Job(TEST_USER, attrs=a)
+
+        job_id = self.server.submit(job)
+
+        # Check that job is in Q state with valid comment
+        job_comment = "Not Running: Insufficient amount of resource: hbmem"
+        self.server.expect(JOB, {'job_state': 'Q', 'comment':
+                           (MATCH_RE, job_comment)}, id=job_id)
```

Comment:
To match two attributes in a single expect() you need to say attrop=PTL_AND.  It'll match one or the other by default.